CMakeLists.txt

```diff
@@ -49,10 +49,12 @@
 cmake_minimum_required (VERSION 3.1)
 ENABLE_LANGUAGE(CXX)
 
-if(NOT CMAKE_BUILD_TYPE)
+if(NOT CMAKE_BUILD_TYPE AND NOT (MSVC_IDE OR XCODE))
```

Comment:
Those are the multi-configuration generators I know - for those the setting is meaningless and the CMAKE_CONFIGURATION_TYPES variable holds those configurations being generated for the IDE